File: Models/Entities/Ticket.cs
using helpdesk.Models.Enums;

namespace helpdesk.Models.Entities
{
    public class Ticket
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        public TicketStatus Status { get; set; } = TicketStatus.Open;

        public int? AssignedAgentId { get; set; }
        public Ticket() { }
    }
}

File: client/src/app/models/ticket.ts
export type TicketStatus = 'Open' | 'InProgress' | 'Done'

export interface Ticket {
    id: number,
    title: string,
    description: string,
    status: TicketStatus,
    assignedAgentId: number | null
}

export interface CreateTicket {
    title: string,
    description: string
}